tests/system/test_ratelimiter.py

```diff
@@ -0,0 +1,77 @@
+from apmserver import BaseTest
+import json
+import os
+import requests
+from nose.plugins.skip import Skip, SkipTest
+import multiprocessing as mp
+import time
+import shutil
+
+
+class Test(BaseTest):
+
+    def test_ratelimiter(self):
+        """
+        Test rate limiting feature.
+        """
+
+        # This test is currently skipped as for testing the rate limiter requests would have to be made in parallel
+        #raise SkipTest
+
+        self.render_config_template(
+            path=os.path.abspath(self.working_dir) + "/log/*",
+            ssl_enabled=False,
+            ratelimit_size=1,
+            ratelimit_burst=1,
+        )
+
+        shutil.copy(self.beat_path + "/fields.yml", self.working_dir)
+
+        proc = self.start_beat()
+        self.wait_until(lambda: self.log_contains("apm-server start running"))
```

Comment:
I remember there was an issue with that when I did the code the first time, but that was quite some time ago and since then lots of things have changed. Will try to change it to ServerBase and see if I hit issues again.